greedy_methods/longest_chunked_palindrome_decomposition.py

```diff
@@ -0,0 +1,82 @@
+"""
+Author  : Alexander Pantyukhin
+Date    : December 16, 2022
+
+Task:
+
+Given a string text. You should split it to k substrings
+(subtext1, subtext2, ..., subtextk) such that:
+
+ - subtexti is a non-empty string.
+
+ - The concatenation of all the substrings is equal to text
+ (i.e., subtext1 + subtext2 + ... + subtextk == text).
+
+ - subtexti == subtextk - i + 1 for all valid values
+ of i (i.e., 1 <= i <= k).
+
+Return the largest possible value of k.
+
+Leetcode reference:
+https://leetcode.com/problems/longest-chunked-palindrome-decomposition/description/
+
+Implementation notes:
+Use greedy approach. If it's possible to find the palindrome
+chunk for the current text start, then use it and apply recursion
+for the left path of string.
+
+Runtime: O(n*n)
+Space: O(1)
+"""
+
+
+def longest_decomposition(text: str) -> int:
+    """
+    >>> longest_decomposition('ghiabcdefhelloadamhelloabcdefghi')
+    7
+    >>> longest_decomposition('merchant')
+    1
+    >>> longest_decomposition('m')
+    1
+    >>> longest_decomposition('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')
+    46
+    >>> longest_decomposition('abcabcabcabcabcabcabcabcabcabcabcabcabcabcabc')
+    15
+    """
+
+    if text is None or len(text) == 0:
```

Comment:
What is text is False or [] or {} or (,) or set()?
```suggestion
    if not text:
```